new code

```diff
@@ -0,0 +1,163 @@
+import java.util.*;
+public class tictactoenew
+{
+    public static void main(String args[])
+    { 
+        String rerun = "yes";
+
+        while(rerun.equalsIgnoreCase("yes"))
+        {
+            System.out.print("welcome!");
+
+            int i,j,k=1,l,m;
```

Comment:
we should declare all the variables on the top. Declare where its needed
eg
int i can move inside for loop
and we can do something like `String player1 =  sc.next();` instead of `player1 = sc.next()`;